From 72e1b49cfa0cfea93577a74fe9cf10ac55f6fe45 Mon Sep 17 00:00:00 2001
From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Fri, 28 Feb 2025 17:06:12 +0800
Subject: [PATCH] tests: Vary the gdctl path used for installed tests

So they don't rely on the source tree.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4309>
(cherry picked from commit c87bf90471730872038b0ee438c8f16c66c62728)
---
 src/tests/meson.build | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/tests/meson.build b/src/tests/meson.build
index a91dfb9893..3cc81f3630 100644
--- a/src/tests/meson.build
+++ b/src/tests/meson.build
@@ -288,6 +288,7 @@ test_cases += [
     'suite': 'backend',
     'sources': [ 'monitor-dbus-tests.c', ],
     'args': gdctl.full_path(),
+    'installed_args': bindir / 'gdctl',
   },
   {
     'name': 'stage-views',
@@ -1050,6 +1051,12 @@ if have_installed_tests
       endforeach
     endif
 
+    if test_case.has_key('installed_args')
+      args = test_case.get('installed_args', [])
+    else
+      args = test_case.get('args', [])
+    endif
+
     installed_tests_cdata = configuration_data()
     installed_tests_cdata.set('apiversion', libmutter_api_version)
     installed_tests_cdata.set('libexecdir', libexecdir)
@@ -1058,7 +1065,7 @@ if have_installed_tests
     installed_tests_cdata.set('gdctltestresultdir', '/tmp/mutter-gdctl-test-results')
     installed_tests_cdata.set('testname', test_case['name'])
     installed_tests_cdata.set('testexecutable', 'mutter-' + test_case['name'])
-    installed_tests_cdata.set('testargs', ' '.join(test_case.get('args', [])))
+    installed_tests_cdata.set('testargs', ' '.join(args))
     installed_tests_cdata.set('runnerargs', ' '.join(runner_args))
 
     configure_file(
-- 
2.50.1

